contract/views/contract.xml

```diff
@@ -85,6 +87,9 @@
             <field name="model">account.analytic.account</field>
             <field name="inherit_id" ref="analytic.view_account_analytic_account_search"/>
             <field name="arch" type="xml">
+                <field name="partner_id" position="attributes">
+                    <attribute name="operator">child_of</attribute>
```

Comment:
But that's not the problem. The problem is that you are modifying the behavior of this search in all the account analytic views. Another alternative is to make a primary view and put it only on contract menu action, not in the rest.